src/main/kotlin/com/coder_rangers/mobius_api/services/implementations/PatientService.kt

```diff
@@ -45,6 +46,16 @@ class PatientService @Autowired constructor(
         return mentalTestService.getPatientTestResult(id)
     }
 
+    override fun cleanTestProgress(id: Long) {
+        val patient = getActivePatientById(id)
+
+        patient.testStatus = IN_PROGRESS
+
+        patient.taskResults?.clear()
+
+        patientDAO.saveOrUpdate(patient)
```

Comment:
Te tiro una solución en una sola línea, pero por si querés aplicarlo. Solo para que la conozcas. Por mí, esto está bien. Pero es así...

```suggestion
        getActivePatientById(id).also { patient ->
           patient.testStatus = IN_PROGRESS
           patient.taskResults?.clear()
           patientDAO.saveOrUpdate(patient)
        }
```